File: Hahn Softwareentwicklung/Models/OrderModel.cs
namespace Hahn_Softwareentwicklung.Models
{
    public class OrderModel
    {
        public Guid OrderId { get; set; }
        public BuyerModel Buyer { get; set; }
        public DateTime OrderDate { get; set; }
        public IList<OrderItemModel> OrderItems { get; set; }
        public decimal TotalAmount { get; set; }
        public PaymentModel Payment { get; set; }
        public OrderStatusModel Status { get; set; }
        public ShippingAddressModel ShippingAddress { get; set; }

        public OrderModel(BuyerModel buyer, List<OrderItemModel> orderItems, PaymentModel payment, ShippingAddressModel shippingAddress)
        {
            OrderId = Guid.NewGuid();
            Buyer = buyer;
            OrderItems = orderItems;
            OrderDate = DateTime.Now;
            Payment = payment;
            TotalAmount = orderItems.Sum(x => x.UnitPrice * x.Quantity);
            Status = OrderStatusModel.Pending;
            ShippingAddress = shippingAddress;
        }

        public void UpdateStatus(OrderStatusModel newStatus)
        {
            Status = newStatus;
        }
    }

    public enum OrderStatusModel
    {
        Pending,
        Processing,
        Shipped,
        Delivered,
        Cancelled
    }
}

File: Angular/Hahn_Softwareentwicklung/src/app/models/order.model.ts
import { BuyerModel } from './buyer.model';
import { OrderItemModel } from './orderItem.model';
import { PaymentModel } from './payment.model';
import { ShippingAddressModel } from './shippingAddress.model';

export class OrderModel {
  id: string;
  buyer: BuyerModel;
  orderDate: Date;
  orderItems: OrderItemModel[];
  totalAmount: number;
  payments: PaymentModel[];
  status: OrderStatus;
  shippingAddress: ShippingAddressModel;

    constructor(id: string, buyer: BuyerModel, orderDate: Date, orderItems: OrderItemModel[], totalAmount: number, payments: PaymentModel[], status: OrderStatus, shippingAddress: ShippingAddressModel) {
        this.id = id;
        this.buyer = buyer;
        this.orderDate = orderDate;
        this.orderItems = orderItems;
        this.totalAmount = totalAmount;
        this.payments = payments;
        this.status = status;
        this.shippingAddress = shippingAddress;
    }
}

export enum OrderStatus {
    Pending = 'Pending',
    Processing = 'Processing',
    Shipped = 'Shipped',
    Delivered = 'Delivered',
    Cancelled = 'Cancelled',
  }